src/Refinement/RefinedTriangulations.jl

```diff
@@ -0,0 +1,126 @@
+
+
+struct RefinedTriangulation{Dc,Dp,A<:Triangulation{Dc,Dp},B<:RefinedDiscreteModel} <: Triangulation{Dc,Dp}
+  trian::A
+  model::B
+end
+
+function get_refined_model(t::RefinedTriangulation)
+  return t.model
+end
+
+# Wrap Triangulation API
+function Geometry.get_background_model(t::RefinedTriangulation)
+  get_background_model(t.trian)
+end
+
+function Geometry.get_grid(t::RefinedTriangulation)
+  get_grid(t.trian)
+end
+
+function Geometry.get_glue(t::RefinedTriangulation,::Val{d}) where d
+  get_glue(t.trian,Val(d))
+end
+
+function Base.view(t::RefinedTriangulation,ids::AbstractArray)
+  v = view(t.trian,ids)
+  return RefinedTriangulation(v,t.model)
+end
+
+# Wrap constructors for RefinedDiscreteModel
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::RefinedDiscreteModel,filter::AbstractArray) where d
+  
+  trian = Triangulation(ReferenceFE{d},get_model(model),filter)
+  return RefinedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::RefinedDiscreteModel,labels::FaceLabeling;kwargs...) where d
+  trian = Triangulation(ReferenceFE{d},get_model(model),labels;kwargs...)
+  return RefinedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(trian::RefinedTriangulation,args...;kwargs...)
+  return RefinedTriangulation(Triangulation(trian.trian),trian.model)
+end
+
+# Domain changes
+# TODO: This assumes we have the same type of triangulation on both refinement levels!
+#       we might want to change this in the future when doing hybrid methods etc...
+
+function Geometry.is_change_possible(strian::RefinedTriangulation,ttrian::RefinedTriangulation)
+  (strian === ttrian) && (return true)
+  smodel = get_refined_model(strian)
+  tmodel = get_refined_model(ttrian)
+  a = get_parent(tmodel) === smodel # tmodel = refine(smodel)
+  b = get_parent(smodel) === tmodel # smodel = refine(tmodel)
+  return a || b
+end
+
+function Geometry.is_change_possible(strian::RefinedTriangulation,ttrian::T) where {T <: Triangulation}
+  smodel = get_refined_model(strian)
+  tmodel = get_background_model(ttrian)
+  a = get_model(smodel) === tmodel  # It is fundamentally the same model
+  b = get_parent(smodel) === tmodel # smodel = refine(tmodel)
+  return a || b
+end
+
+function Geometry.is_change_possible(strian::T,ttrian::RefinedTriangulation) where {T <: Triangulation}
+  return is_change_possible(ttrian,strian)
+end
+
+function Geometry.best_target(strian::RefinedTriangulation,ttrian::RefinedTriangulation)
+  @check is_change_possible(strian,ttrian)
+  smodel = get_refined_model(strian)
+  tmodel = get_refined_model(ttrian)
+  a = get_model(smodel) === get_parent(tmodel)
+  b = get_parent(tmodel) === smodel # tmodel = refine(smodel)
```

Comment:
If the parent of `tmodel` is never of type `RefinedDiscreteModel` (see discussion above), this condition can never be true.

Reply:
I agree. I also think this should be the way to go (and then remove this check).

Reply:
Resolved by [0e3c6eb](https://github.com/gridap/Gridap.jl/pull/838/commits/0e3c6eb9cdc1b3dd995a65f7732f940ee853e28b)